usrsctplib/user_environment.c

```diff
@@ -85,6 +85,31 @@ read_random(void *buf, size_t size)
 	return;
 }
 
+void
+finish_random(void)
+{
+	return;
+}
+#elif defined(SCTP_USE_OPENSSL_RAND)
+#include <openssl/rand.h>
+
+void
+init_random(void)
+{
+	return;
+}
+
+void
+read_random(void *buf, size_t size)
+{
+	while (!RAND_bytes((uint8_t *)buf, size)) {
```

Comment:
Do we expect RAND_bytes to succeed on subsequent invocations if the first one didn't return 1?

Reply:
I have no idea. In the FreeBSD sources `arc4random` doesn't fail, therefore there is no code to deal with an error case. I'm not sure if there is always a simple way to signal this to the user, since it might occur during packet handling.

Reply:
Yes, that might be tricky (I am not an expert of the library) and since this is not an API function for usrsctp this might even require a multi-layer error status propagation.

Taylor: I think this version of `read_random` cannot assume that `RAND_bytes` never fails (like it is documented for `arc4random` so we need to find another solution (see my comment below, -1 is an error condition that will silently return in this case).

Reply:
Keep in mind this is probably only going to be used by chromium, where we use boringssl and `RAND_bytes` never fails.

Reply:
Yes, on BoringSSL `RAND_bytes` never fails so if we can enforce BoringSSL here it would be nice (to ensure that users of the standalone WebRTC library don't define `SCTP_USE_OPENSSL_RAND` and also `rtc_ssl_root` in GN.

What do you think about using `OPENSSL_IS_BORINGSSL` (see https://source.chromium.org/chromium/chromium/src/+/master:third_party/boringssl/src/include/openssl/base.h;l=182;drc=ca058c06474c4492f0fbd517679793234dde6165)?

Something like:

```
#include <openssl/base.h>

#ifndef OPENSSL_IS_BORINGSSL
#error Only BoringSSL is supported with SCTP_USE_OPENSSL_RAND.
#endif
```

Reply:
Fine with me. One add  comment why it is enforced.

Reply:
Done.